tests/DotNetLightning.Core.Tests/TransactionTests.fs

```diff
@@ -140,6 +140,84 @@ let testList = testList "transaction tests" [
+                            (currentCommitmentNumber: CommitmentNumber)
+                                : PerCommitmentSecretStore =
+                if currentCommitmentNumber = commitmentNumber then
+                    remoteRemotePerCommitmentSecrets
+                else
+                    let currentPerCommitmentSecret =
+                        localChannelPrivKeys.CommitmentSeed.DerivePerCommitmentSecret
+                            currentCommitmentNumber
+                    let nextLocalPerCommitmentSecretsRes =
+                        remoteRemotePerCommitmentSecrets.InsertPerCommitmentSecret
+                            currentCommitmentNumber
+                            currentPerCommitmentSecret
+                    addKeys
+                        (Result.deref nextLocalPerCommitmentSecretsRes)
+                        (currentCommitmentNumber.NextCommitment())
+            addKeys (PerCommitmentSecretStore()) CommitmentNumber.FirstCommitment
+
+        let remoteRemoteCommit = {
+            Index = commitmentNumber
+            Spec = commitmentSpec
+            TxId = TxId <| commitmentTx.GetHash()
+            RemotePerCommitmentPoint = perCommitmentPoint
+        }
+        let remoteLocalParams: LocalParams = {
+            NodeId = remoteParams.NodeId
+            ChannelPubKeys = remoteParams.ChannelPubKeys
+            DustLimitSatoshis = remoteParams.DustLimitSatoshis
+            MaxHTLCValueInFlightMSat = remoteParams.MaxHTLCValueInFlightMSat
+            ChannelReserveSatoshis = remoteParams.ChannelReserveSatoshis
+            HTLCMinimumMSat = remoteParams.HTLCMinimumMSat
+            ToSelfDelay = remoteParams.ToSelfDelay
+            MaxAcceptedHTLCs = remoteParams.MaxAcceptedHTLCs
+            IsFunder = false
+            DefaultFinalScriptPubKey = remoteDestPubKey.ScriptPubKey
+            Features = remoteParams.Features
+        }
+        let remoteRemoteParams = {
+            NodeId = localParams.NodeId
+            DustLimitSatoshis = localParams.DustLimitSatoshis
+            MaxHTLCValueInFlightMSat = localParams.MaxHTLCValueInFlightMSat
+            ChannelReserveSatoshis = localParams.ChannelReserveSatoshis
+            HTLCMinimumMSat = localParams.HTLCMinimumMSat
+            ToSelfDelay = localParams.ToSelfDelay
+            MaxAcceptedHTLCs = localParams.MaxAcceptedHTLCs
+            ChannelPubKeys = localParams.ChannelPubKeys
+            Features = localParams.Features
+            MinimumDepth = 6u |> BlockHeightOffset32
+        }
+
+        let transactionBuilderOpt =
+            ForceCloseFundsRecovery.tryGetFundsFromRemoteCommitmentTx
+                remoteLocalParams
+                remoteRemoteParams
+                fundingScriptCoin
+                remoteRemotePerCommitmentSecrets
+                remoteRemoteCommit
+                remoteChannelPrivKeys
+                Network.RegTest
+                commitmentTx
+        let transactionBuilder = transactionBuilderOpt.Value
```

Comment:
shouldn't this need a match? otherwise if it's None it could blow up here

Reply:
This is a test so it does not really matter.
It is slightly better to assert by `Expect.isSome` before this line. But it is not a big deal.

Reply:
I added a couple of `Expect.isSome`s.